File: Best_Time_to_Buy_and_Sell_Stock/best_time_to_buy_and_sell_stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int buy = 0;
        int sell = 0;
        int maxReturn = 0;
        while (sell < prices.length) {
            if (prices[buy] < prices[sell]) {
                maxReturn = Math.max(maxReturn, prices[sell] - prices[buy]);
            } else {
                buy = sell;
            }
            sell ++;
        }
        return maxReturn;
    }
}

File: Best_Time_to_Buy_and_Sell_Stock/best_time_to_buy_and_sell_stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy, sell = 0, 0
        max_return = 0
        while sell < len(prices):
            if prices[buy] < prices[sell]:
                max_return = max(max_return, prices[sell] - prices[buy])
            else:
                buy = sell
            sell += 1
        return max_return
